diff --git a/dist/compiler/js/compiler.js b/dist/compiler/js/compiler.js
index 70caef1774fe44b3ebf4778b36b1deabb151fa0a..d3aef3e768aa2948b38009a8bc2fee28e8a6639b 100644
--- a/dist/compiler/js/compiler.js
+++ b/dist/compiler/js/compiler.js
@@ -31,6 +31,7 @@ var cssSideEffectImports = require('../plugins/cssSideEffectImports.js');
 var vanillaExtract = require('../plugins/vanillaExtract.js');
 var invariant = require('../../invariant.js');
 var hmr = require('./plugins/hmr.js');
+var million = require('million/compiler');
 
 function _interopNamespace(e) {
   if (e && e.__esModule) return e;
@@ -93,7 +94,7 @@ const createEsbuildConfig = (ctx, refs) => {
     includeNodeModules: true
   }), esbuildPluginsNodeModulesPolyfill.nodeModulesPolyfillPlugin(), external.externalPlugin(/^node:.*/, {
     sideEffects: false
-  })];
+  }), million.esbuild({ server: true })];
   if (ctx.options.mode === "development" && ctx.config.future.v2_dev) {
     plugins.push(hmr.hmrPlugin(ctx));
   }
diff --git a/dist/compiler/server/compiler.js b/dist/compiler/server/compiler.js
index a5143271bd5f30487d7f2b052ce9b37229add46c..396e023f188ef75ac52539d7876ac4fa6727ff19 100644
--- a/dist/compiler/server/compiler.js
+++ b/dist/compiler/server/compiler.js
@@ -29,6 +29,7 @@ var entry = require('./plugins/entry.js');
 var routes = require('./plugins/routes.js');
 var external = require('../plugins/external.js');
 var cssBundlePlugin = require('../plugins/cssBundlePlugin.js');
+var million = require('million/compiler');
 
 function _interopNamespace(e) {
   if (e && e.__esModule) return e;
@@ -70,7 +71,7 @@ const createEsbuildConfig = (ctx, refs) => {
     sideEffects: false
   }), mdx.mdxPlugin(ctx), emptyModules.emptyModulesPlugin(ctx, /\.client(\.[jt]sx?)?$/), routes.serverRouteModulesPlugin(ctx), entry.serverEntryModulePlugin(ctx), manifest.serverAssetsManifestPlugin(refs), bareImports.serverBareModulesPlugin(ctx), external.externalPlugin(/^node:.*/, {
     sideEffects: false
-  })];
+  }), million.esbuild({ server: true })];
   if (ctx.config.serverPlatform !== "node") {
     plugins.unshift(esbuildPluginsNodeModulesPolyfill.nodeModulesPolyfillPlugin());
   }